daemons/ipa-slapi-plugins/ipa-pwd-extop/ipa_pwd_extop.c

```diff
@@ -561,16 +561,21 @@ static int ipapwd_chpwop(Slapi_PBlock *pb, struct ipapwd_krbcfg *krbcfg)
 	/* check the policy */
 	ret = ipapwd_CheckPolicy(&pwdata);
 	if (ret) {
-		errMesg = ipapwd_error2string(ret);
 		if (ret == IPAPWD_POLICY_ERROR) {
 			errMesg = "Internal error";
 			rc = ret;
-		} else {
+			goto free_and_return;
+		}
+		/* ipapwd_CheckPolicy happily will try to apply a policy
+		 * even if it doesn't need to be applied for Directory Manager
+		 * or passsync managers, filter that error out */
+		if (pwdata.changetype != IPA_CHANGETYPE_DSMGR) {
+			errMesg = ipapwd_error2string(ret);
 			ret = ipapwd_to_ldap_pwpolicy_error(ret);
 			slapi_pwpolicy_make_response_control(pb, -1, -1, ret);
 			rc = LDAP_CONSTRAINT_VIOLATION;
+			goto free_and_return;
 		}
-		goto free_and_return;
```

Comment:
This is what I did in the first variant of this patch but then I realized the point of ipapwd_CheckPolicy() is also to update the password history. If we are applying a password history for non-Kerberos accounts, we should call ipapwd_CheckPolicy() so that ipapwd_check_policy() would update the history.

Reply:
Yeah then it's time we split Checks from Updates, it's hard to reason about an interface when it starts doing things it does not declare.